### PR TITLE
Dev digest of counts

### PR DESCRIPTION
**Pull request by manu-govind**
Enhanced Notifications: Digest of Counts implementation using commons-db 
Also renamed a class in subscription-management lib
Moved out some classes in to their own files in the notification-workflow ktor.

### CODE REVIEW

File: pstatus-notifications-workflow-ktor/build.gradle.kts [hunk 1]
```diff
@@ -33,6 +33,9 @@ application {
 
 dependencies {
     implementation(project(":libs:commons-database"))
+    implementation("software.amazon.awssdk:dynamodb:2.17.98")
```
Comment (mkrystof) [inline pstatus-notifications-workflow-ktor/build.gradle.kts:36]: We should be able to removed these 3 lines for dynamodb.

File: pstatus-notifications-workflow-ktor/build.gradle.kts [hunk 2]
```diff
@@ -62,6 +65,8 @@ dependencies {
     implementation ("io.ktor:ktor-client-content-negotiation:2.1.0")
     implementation ("io.netty:netty-all:4.1.68.Final")
     implementation ("io.netty:netty-tcnative-boringssl-static:2.0.52.Final:windows-x86_64")
+    implementation("software.amazon.awssdk:dynamodb-enhanced:2.20.70") // Latest Enhanced Client for DynamoDB
```
Comment (mkrystof) [inline pstatus-notifications-workflow-ktor/build.gradle.kts:68]: Should be able to remove these two lines as well for dynamodb.

File: pstatus-notifications-workflow-ktor/src/main/kotlin/Application.kt [hunk 3]
```diff
@@ -1,28 +1,51 @@
...
 import org.koin.ktor.plugin.Koin
 
 fun KoinApplication.loadKoinModules(environment: ApplicationEnvironment): KoinApplication {
-    val cosmosModule = module {
-        val uri = environment.config.property("azure.cosmos_db.client.endpoint").getString()
-        val authKey = environment.config.property("azure.cosmos_db.client.key").getString()
-        single<ProcessingStatusRepository>{ CosmosRepository(uri, authKey) } //createdAtStart = true is not working , with lib integration
-        // Create a CosmosDB config that can be dependency injected (for health checks)
-        single(createdAtStart = true) { CosmosConfiguration(uri, authKey) }
+    val logger = KotlinLogging.logger {}
```
Comment (mkrystof) [inline pstatus-notifications-workflow-ktor/src/main/kotlin/Application.kt:23]: Please replace lines 24 thru 46 with:
```kotlin
val databaseModule = DatabaseKoinCreator.moduleFromAppEnv(environment)
```

**Review by mkrystof — CHANGES_REQUESTED** [review]

File: pstatus-notifications-workflow-ktor/build.gradle.kts (hunk 2 above)
Comment (manu-govind) [reply pstatus-notifications-workflow-ktor/build.gradle.kts:65]: This is required for the dynamo db annotation for the response class. The other one has been removed.

**Review by mkrystof — APPROVED** [review]
lgtm

**Pull request merged**